[package] libtiff/4.3.0: local conan create fails with cmake configuration error

<!-- 
  Please don't forget to update the issue title.
  Include all applicable information to help us reproduce your problem.
-->

### Package and Environment Details (include every applicable attribute)
  * Package Name/Version: **libtiff/4.3.0**
  * Operating System+version: **Linux Ubuntu 18.04**
  * Compiler+version: **GCC 7**
  * Docker image: 
  * Conan version: **conan 1.46.2**
  * Python version: **Python 3.6.9**


### Conan profile (output of `conan profile show default` or `conan profile show <profile>` if custom profile is in use)
```
[settings]
arch=x86_64
arch.march=None
arch_build=x86_64
build_type=Release
compiler=gcc
compiler.libcxx=libstdc++11
compiler.version=7
os=Linux
os_build=Linux
[options]
[build_requires]
[env]
```


### Steps to reproduce (Include if Applicable)
`conan create recipes/libtiff/all/conanfile.py 4.3.0@`


### Logs (Include/Attach if Applicable)
<details><summary>Click to expand log</summary>

```
-- Conan: Using autogenerated Findjbig.cmake
-- Found jbig: 20160605 (found version "20160605") 
-- Library jbig found /home/usr/.conan/data/jbig/20160605/_/_/package/d8fff784e71b40316e752103c83b4698d094b5e3/lib/libjbig.a
-- Found: /home/usr/.conan/data/jbig/20160605/_/_/package/d8fff784e71b40316e752103c83b4698d094b5e3/lib/libjbig.a
CMake Error at source_subfolder/cmake/FindJBIG.cmake:4 (target_link_libraries):
  Cannot specify link libraries for target "JBIG::JBIG" which is not built by
  this project.
Call Stack (most recent call first):
  source_subfolder/cmake/JBIGCodec.cmake:30 (find_package)
  source_subfolder/CMakeLists.txt:101 (include)


-- Configuring incomplete, errors occurred!
See also "/home/usr/.conan/data/libtiff/4.3.0/_/_/build/8311e1632d035c91234ae06c2588dd1f1f1da4e6/build_subfolder/CMakeFiles/CMakeOutput.log".
See also "/home/usr/.conan/data/libtiff/4.3.0/_/_/build/8311e1632d035c91234ae06c2588dd1f1f1da4e6/build_subfolder/CMakeFiles/CMakeError.log".
libtiff/4.3.0: 
libtiff/4.3.0: ERROR: Package '8311e1632d035c91234ae06c2588dd1f1f1da4e6' build failed
libtiff/4.3.0: WARN: Build folder /home/usr/.conan/data/libtiff/4.3.0/_/_/build/8311e1632d035c91234ae06c2588dd1f1f1da4e6
ERROR: libtiff/4.3.0: Error in build() method, line 181
	cmake = self._configure_cmake()
while calling '_configure_cmake', line 176
	self._cmake.configure(build_folder=self._build_subfolder)
	ConanException: Error 1 while executing cd '/home/usr/.conan/data/libtiff/4.3.0/_/_/build/8311e1632d035c91234ae06c2588dd1f1f1da4e6/build_subfolder' && cmake -G "Unix Makefiles" -DCMAKE_BUILD_TYPE="Release" -DCONAN_IN_LOCAL_CACHE="ON" -DCONAN_COMPILER="gcc" -DCONAN_COMPILER_VERSION="7" -DCONAN_CXX_FLAGS="-m64" -DCONAN_SHARED_LINKER_FLAGS="-m64" -DCONAN_C_FLAGS="-m64" -DCONAN_LIBCXX="libstdc++11" -DBUILD_SHARED_LIBS="OFF" -DCMAKE_INSTALL_PREFIX="/home/usr/.conan/data/libtiff/4.3.0/_/_/package/8311e1632d035c91234ae06c2588dd1f1f1da4e6" -DCMAKE_INSTALL_BINDIR="bin" -DCMAKE_INSTALL_SBINDIR="bin" -DCMAKE_INSTALL_LIBEXECDIR="bin" -DCMAKE_INSTALL_LIBDIR="lib" -DCMAKE_INSTALL_INCLUDEDIR="include" -DCMAKE_INSTALL_OLDINCLUDEDIR="include" -DCMAKE_INSTALL_DATAROOTDIR="share" -DCONAN_CMAKE_POSITION_INDEPENDENT_CODE="ON" -DCMAKE_MODULE_PATH="/home/usr/.conan/data/libtiff/4.3.0/_/_/build/8311e1632d035c91234ae06c2588dd1f1f1da4e6" -DCMAKE_EXPORT_NO_PACKAGE_REGISTRY="ON" -DCONAN_EXPORTED="1" -Dlzma="True" -Djpeg="True" -Djbig="True" -Dzlib="True" -Dlibdeflate="True" -Dzstd="True" -Dwebp="True" -Dcxx="True" -Wno-dev '/home/usr/.conan/data/libtiff/4.3.0/_/_/build/8311e1632d035c91234ae06c2588dd1f1f1da4e6'

```

</details>

--- a/recipes/libtiff/all/patches/4.3.0-0001-cmake-add-libjpeg-turbo.patch
+++ b/recipes/libtiff/all/patches/4.3.0-0001-cmake-add-libjpeg-turbo.patch
@@ -5,9 +5,9 @@
 +if(NOT TARGET ZSTD::ZSTD)
 +  add_library(ZSTD::ZSTD INTERFACE IMPORTED)
 +  if(TARGET zstd::libzstd_shared)
-+    target_link_libraries(ZSTD::ZSTD INTERFACE zstd::libzstd_shared)
++    set_target_properties(ZSTD::ZSTD PROPERTIES INTERFACE_LINK_LIBRARIES zstd::libzstd_shared)
 +  else()
-+    target_link_libraries(ZSTD::ZSTD INTERFACE zstd::libzstd_static)
++    set_target_properties(ZSTD::ZSTD PROPERTIES INTERFACE_LINK_LIBRARIES zstd::libzstd_static)
 +  endif()
 +endif()
 --- cmake/FindJBIG.cmake
@@ -16,7 +16,7 @@
 +include("${cmake_wrapper_SOURCE_DIR}/ConanFindjbig.cmake")
 +if(NOT TARGET JBIG::JBIG)
 +  add_library(JBIG::JBIG INTERFACE IMPORTED)
-+  target_link_libraries(JBIG::JBIG INTERFACE jbig::jbig)
++  set_target_properties(JBIG::JBIG PROPERTIES INTERFACE_LINK_LIBRARIES jbig::jbig)
 +endif()
 --- cmake/JPEGCodec.cmake
 +++ cmake/JPEGCodec.cmake

